File: java/program55.java
import java.util.*;

class ArrayX
{
    public int Arr[];

    ArrayX(int iSize)
    {
        this.Arr = new int[iSize];
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements of array :");

        for(int i = 0; i < Arr.length; i++)
        {
            Arr[i] = sobj.nextInt();
        }
    }
    
    public void Display()
    {
        System.out.println("Elements of the array are : ");

        for(int i = 0; i < Arr.length; i++)
        {
            System.out.print(Arr[i]+"\t");
        }
        System.out.println();
    }
}

class MyArray extends ArrayX
{
    MyArray(int size)
    {
        super(size);
    }

    public int Minimum()
    {
        int iCnt = 0;
        int Min = Arr[iCnt];

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] < Min)
            {
                Min = Arr[iCnt];
            }
        }
        return Min;
    }

    public int Maximum()
    {
        int iCnt = 0;
        int Max = Arr[iCnt];

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] > Max)
            {
                Max = Arr[iCnt];
            }
        }
        return Max;   
    }
}

class program55
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements");
        int size = sobj.nextInt();

        MyArray obj = new MyArray(size);
        obj.Accept();
        obj.Display();

        System.out.println("Maximum element of the array is : "+obj.Maximum());
        System.out.println("Minimum element of the array is : "+obj.Minimum());
    }
}

File: cpp/program55.cpp
// program85.c

#include<iostream>
using namespace std;

class ArrayX
{
    protected:
        int *Arr;
        int iSize;
        int iCnt;

    public:

        ArrayX(int iLength)
        {
            iCnt = 0;
            iSize = iLength;
            Arr = new int[iSize];
        }
    
    protected:

        void Accept()
        {
            cout<<"Enter elements of array :\n";

            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                cin>>Arr[iCnt];
            }
        }

        void Dislay()
        {
            cout<<"Elements of the array are :\n";

            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                cout<<Arr[iCnt]<<"\t";
            }
            cout<<"\n";
        }
};

class MyArray : public ArrayX
{
    public:

        MyArray(int iSize) : ArrayX(iSize)
        {
            ArrayX::Accept();
            ArrayX::Dislay();
        }

        int iMaximum()
        {
            int iCnt = 0;
            int iMax = Arr[0];

            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                if(iMax < Arr[iCnt])
                {
                    iMax = Arr[iCnt];
                }
            }
            return iMax;
        }

        int iMinimum()
        {
            int iMin = Arr[0];

            for(int iCnt = 0; iCnt < iSize; iCnt++)
            {
                if(iMin > Arr[iCnt])
                {
                    iMin = Arr[iCnt];
                }
            }
            return iMin;
        }
};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter number of elements \n";
    cin>>iValue;
    
    MyArray obj(iValue);

    iRet = obj.iMaximum();
    cout<<"Maximum number in array is : "<<iRet<<"\n";

    iRet = obj.iMinimum();
    cout<<"Minimum number in array is : "<<iRet<<"\n";

    return 0;
}